src/functions.jl

```diff
@@ -6,649 +6,646 @@
-    sincoefhom = zero(T)
-    coscoefhom = zero(T)
+        function sinhcosh!(s::$T, c::$T, a::$T, k::Int)
+            if $T == Taylor1
+                if k == 0
+                    @inbounds s[1] = asinh( a[1] )
+                    @inbounds c[1] = acosh( a[1] )
+                    return nothing
+                end
+                @inbounds s[k+1] = zero(s[1])
+                @inbounds c[k+1] = zero(c[1])
+            else
+                if k == 0
+                    @inbounds s[1] = asinh( a[1][1] )
+                    @inbounds c[1] = acosh( a[1][1] )
+                    return nothing
+                end
+                @inbounds s[k+1] = HomogeneousPolynomial(zero(s[1][1]), k)
+                @inbounds c[k+1] = HomogeneousPolynomial(zero(c[1][1]), k)
+            end
+
+            @inbounds for i = 1:k
+                x = i * a[i+1]
+                s[k+1] += x * c[k-i+1]
+                c[k+1] += x * s[k-i+1]
+            end
+            s[k+1] = s[k+1]/k
+            c[k+1] = c[k+1]/k
+            return nothing
+        end
 
-    @inbounds for i = 1:kcoef
-        x = i * ac[i+1]
-        sincoefhom += x * ccoeffs[kcoef-i+1]
-        coscoefhom -= x * scoeffs[kcoef-i+1]
+        function tanh!(c::$T, a::$T, c2::$T, k::Int)
+            if $T == Taylor1
+                if k == 0
+                    @inbounds aux = tanh( a[1] )
+                    @inbounds c[1] = aux
+                    @inbounds c2[1] = aux^2
+                    return nothing
+                end
+                @inbounds c[k+1] = zero(c[1])
+            else
+                if k == 0
+                    @inbounds aux = tanh( a[1][1] )
+                    @inbounds c[1] = aux
+                    @inbounds c2[1] = aux^2
+                    return nothing
+                end
+                @inbounds c[k+1] = HomogeneousPolynomial(zero(c[1][1]), k)
+            end
+
+            @inbounds for i = 0:k-1
+                c[k+1] += (k-i)*a[k-i+1]*c2[i+1]
+            end
+            @inbounds c[k+1] = a[k+1] - c[k+1]/k
+            sqr!(c2, c, k)
+
+            return nothing
+        end
     end
-
-    sincoefhom = sincoefhom/kcoef
-    coscoefhom = coscoefhom/kcoef
-    return sincoefhom, coscoefhom
 end
 
-## tan ##
-doc"""
-    tan(a)
-
-Return the Taylor expansion of $\tan(a)$, of order `a.order`, for
-`a::Taylor1` polynomial.
 
-For details on making the Taylor expansion, see
-[`TaylorSeries.tanHomogCoef`](@ref).
-"""
-function tan(a::Taylor1)
-    aux = tan( a[1] )
-    T = typeof(aux)
-    v = convert(Array{T,1}, a.coeffs)
-    coeffs = similar(v)
-    coeffst2 = similar(v)
-    @inbounds coeffs[1] = aux
-    @inbounds coeffst2[1] = aux^2
-    @inbounds for k = 1:a.order
-        coeffs[k+1] = tanHomogCoef(k, v, coeffst2)
-        coeffst2[k+1] = squareHomogCoef(k, coeffs)
-    end
-    Taylor1( coeffs, a.order )
-end
-
-# Homogeneous coefficients for tan
 doc"""
-    tanHomogCoef(kcoef, ac, coeffst2)
+    reverse(f)
```

Comment:
I'm ok with whatever term you choose. For the case that is implemented (which is numerically stable), I believe inversion and reversion are equivalent.